### PR TITLE
Certs names should be taken dynamically in indexer-security-init.sh

### PR DESCRIPTION
**Pull request by miguelfdez99**
|Related issue|
|---|
|#1409|

## Description
Certificates names were not taken dynamically, this had to be changed.

In order to change this problem, the following things had been done:
- wazuh_indexer_root_ca is a variable which take root-ca.pem from the file opensearch.yml
-  WAZUH_INDEXER_ADMIN_PATH contains the directory where root-ca is stored.
- If admin.pem and admin-key.pem are in the same directory as root-ca it uses them.
- If not it will launch an error message giving the command to generate the keys.

## Logs example
I'm testing different use cases:
- 1: admin.pem does not exist

- 2: admin-key.pem does not exist

- 3: root-ca.pem does not exist

- 4: wazuh-indexer it's not up

- 5: Using port 9650 instead of 9300

- 6: network.host it's 0.0.0.0

- 7: network.host it's a normal internal ip
Random private IP

IP from the other node

- 8: network.host name it's an interface 

 9: /etc/wazuh-indexer/opensearch.yml does not exist

- 10: SECURITY_PATH does not exist

- 11: INSTALL_PATH does not exist

- 12: CONFIG_PATH does not exist

- 13: JAVA_HOME="${INSTALL_PATH}/jdk does not exist
It does not show an error but neither that it worked.

### CODE REVIEW

File: stack/indexer/indexer-security-init.sh [hunk 1]
```diff
@@ -42,11 +42,17 @@ fi
...
 # -----------------------------------------------------------------------------
 
+eval "wazuh_indexer_root_ca=( $(cat /etc/wazuh-indexer/opensearch.yml | grep plugins.security.ssl.transport.pemtrustedcas_filepath | sed 's/plugins.security.ssl.transport.pemtrustedcas_filepath:\s//') )"
+WAZUH_INDEXER_ADMIN_PATH="${wazuh_indexer_root_ca%%/root-ca.*}"
 securityadmin() {
 SECURITY_PATH="${INSTALL_PATH}/plugins/opensearch-security"
 
-OPENSEARCH_PATH_CONF="${CONFIG_PATH}" JAVA_HOME="${INSTALL_PATH}/jdk" runuser wazuh-indexer --shell="/bin/bash" --command="${SECURITY_PATH}/tools/securityadmin.sh -cd ${SECURITY_PATH}/securityconfig -cacert ${CONFIG_PATH}/certs/root-ca.pem -cert ${CONFIG_PATH}/certs/admin.pem -key ${CONFIG_PATH}/certs/admin-key.pem -h ${HOST} -p ${PORT} ${OPTIONS}"
-
+if [ -f "${WAZUH_INDEXER_ADMIN_PATH}/admin.pem" ] && [ -f "${WAZUH_INDEXER_ADMIN_PATH}/admin-key.pem" ]; then
```
Comment (okynos) [inline stack/indexer/indexer-security-init.sh:50]: Check also if root CA is present.

File: stack/indexer/indexer-security-init.sh [hunk 2]
```diff
@@ -42,11 +42,17 @@ fi
 
 # -----------------------------------------------------------------------------
 
+eval "wazuh_indexer_root_ca=( $(cat /etc/wazuh-indexer/opensearch.yml | grep plugins.security.ssl.transport.pemtrustedcas_filepath | sed 's/plugins.security.ssl.transport.pemtrustedcas_filepath:\s//') )"
```
Comment (okynos) [inline stack/indexer/indexer-security-init.sh:45]:
```suggestion
wazuh_indexer_root_ca=$(cat /etc/wazuh-indexer/opensearch.yml | grep http.pemtrustedcas | sed 's/.*: //')"
```

**Review by okynos — CHANGES_REQUESTED** [review]
Only a few changes 👍🏼

File: stack/indexer/indexer-security-init.sh (hunk 1 above)
Comment (miguelfdez99) [reply stack/indexer/indexer-security-init.sh]: Done! https://github.com/wazuh/wazuh-packages/commit/80727a55d4c3d650f180e3c67492d579c147a5f8

File: stack/indexer/indexer-security-init.sh (hunk 2 above)
Comment (miguelfdez99) [reply stack/indexer/indexer-security-init.sh]: Done! https://github.com/wazuh/wazuh-packages/commit/80727a55d4c3d650f180e3c67492d579c147a5f8

**Review by okynos — APPROVED** [review]
LGTM

File: stack/indexer/indexer-security-init.sh [hunk 3]
```diff
@@ -42,11 +42,17 @@ fi
 
 # -----------------------------------------------------------------------------
 
+WAZUH_INDEXER_ROOT_CA=$(cat /etc/wazuh-indexer/opensearch.yml | grep http.pemtrustedcas | sed 's/.*: //')"
+WAZUH_INDEXER_ADMIN_PATH="${WAZUH_INDEXER_ROOT_CA%%/root-ca.*}"
```
Comment (DFolchA) [inline stack/indexer/indexer-security-init.sh:46]: What if the key is not named root-ca??
you can use `dirname $file` to obtain the path to the directory

**Review by DFolchA — CHANGES_REQUESTED** [review]

File: stack/indexer/indexer-security-init.sh (hunk 3 above)
Comment (miguelfdez99) [reply stack/indexer/indexer-security-init.sh]: Done https://github.com/wazuh/wazuh-packages/commit/59c81ff61ec2d872a466910d634b90ccef023e6a

**Review by alberpilot — CHANGES_REQUESTED** [review]
- The `securityadmin` function has a wrong indentation
- The `SECURITY_PATH` variable can be placed after OPTIONS at the top. 
- It's needed to ensure that all checks output are null redirected. If not, this could be an output:
```
grep: /etc/wazuh-indexer/opensearch.yml: No such file or directory
grep: /etc/wazuh-indexer/opensearch.yml: No such file or directory
cat: /etc/wazuh-indexer/opensearch.yml: No such file or directory
dirname: missing operand
Try 'dirname --help' for more information.
ERROR: admin.pem and admin-key.pem could not be found in . You must execute the following command: JAVA_HOME=/usr/share/wazuh-indexer/jdk runuser wazuh-indexer --shell=/bin/bash --command=/usr/share/wazuh-indexer/plugins/opensearch-security/tools/securityadmin.sh -cd /usr/share/wazuh-indexer/plugins/opensearch-security/plugins/opensearch-security/securityconfig -cacert  -cert $/etc/wazuh-indexer/certs/admin.pem -key /etc/wazuh-indexer/certs/admin-key.pem -h  -p 9300 -icl -nhnv 
```
If some check failed, I only want to see the error message. 
- The sentence must be re-phrased by: ERROR: this tool try to find admin.pem and admin-key.pem in ${WAZUH_INDEXER_ADMIN_PATH} but it couldn't. In this case, you must run manually the Indexer security initializer by running the command: JAVA_HOME="/usr/share/wazuh-indexer/jdk" runuser wazuh-indexer --shell="/bin/bash" --command="/usr/share/wazuh-indexer/plugins/opensearch-security/tools/securityadmin.sh -cd /usr/share/wazuh-indexer/plugins/opensearch-security/plugins/opensearch-security/securityconfig -cacert /path/to/root-ca.pem -cert /path/to/admin.pem -key /path/to/admin-key.pem -h ${HOST} -p ${PORT} ${OPTIONS}" replacing `/path/to/` by your certificates path."
- Why `$` in `/etc/wazuh-indexer/certs/admin.pem` path?

**Review by alberpilot — CHANGES_REQUESTED** [review]
Based on your provided user case testing:
1. Message expected. :heavy_check_mark: 
2. Message expected. :heavy_check_mark: 
3. Message expected. :heavy_check_mark: 
4. Message expected. :heavy_check_mark: 
5. Message expected. :heavy_check_mark: 
6. If network.host = 0.0.0.0, the script should use 127.0.0.1, as you can see in the code, getNetworkHost function. Please investigate why. :red_circle: 
7. Message expected. :heavy_check_mark: 
8. If network.host = eth0, the script should use the IP associated to the interface, as getNetworkFunction looks to be defining. Please investigate why. :red_circle: 
9. If `/etc/wazuh-indexer/opensearch.yml` doesn't exist, the script must show a message saying: `it was not possible to find /etc/wazuh-indexer/opensearch.yml`. :red_circle: 
10. Same. :red_circle: 
11. Same. :red_circle: 
12. The script should be able to control de error message. :red_circle: 
13. The script should be able to show an error message if JAVA_HOME doesn't exist. :red_circle: 
14.

**Comment by miguelfdez99** [comment]
- 6:  network.host it's 0.0.0.0


- 8: network.host name it's an interface
I've tried several IPs and interface names and everything works as expected. I've changed the function getNetworkHost() in order to achieve this. https://github.com/wazuh/wazuh-packages/pull/1441/commits/df46a66ea69a2530d7e3f6d0dc139d742f5e2beb


- https://github.com/wazuh/wazuh-packages/pull/1441/commits/74009526f390e74188438370ec57da99dece5ab9
In this commit I have added several error messages that could happen.

- 9: /etc/wazuh-indexer/opensearch.yml does not exist


- 10: SECURITY_PATH does not exist


- 11: INSTALL_PATH does not exist


- 12: CONFIG_PATH does not exist


- 13: JAVA_HOME="${INSTALL_PATH}/jdk does not exist

File: stack/indexer/indexer-security-init.sh [hunk 4]
```diff
@@ -12,40 +12,76 @@ HOST=""
 INSTALL_PATH="/usr/share/wazuh-indexer"
 CONFIG_PATH="/etc/wazuh-indexer"
 OPTIONS="-icl -nhnv"
+WAZUH_INDEXER_ROOT_CA="$(cat /etc/wazuh-indexer/opensearch.yml 2>&1 | grep http.pemtrustedcas | sed 's/.*: //')"
+WAZUH_INDEXER_ADMIN_PATH="$(dirname "${WAZUH_INDEXER_ROOT_CA}" 2>&1)"
+SECURITY_PATH="${INSTALL_PATH}/plugins/opensearch-security"
+CONFIG_FILE="${CONFIG_PATH}/opensearch.yml"
```
Comment (okynos) [inline stack/indexer/indexer-security-init.sh:18]: Set up the first and use in the rest of calls e.g. `L15`

File: stack/indexer/indexer-security-init.sh [hunk 5]
```diff
@@ -12,40 +12,76 @@ HOST=""
...
 
 securityadmin() {
-SECURITY_PATH="${INSTALL_PATH}/plugins/opensearch-security"
 
-OPENSEARCH_PATH_CONF="${CONFIG_PATH}" JAVA_HOME="${INSTALL_PATH}/jdk" runuser wazuh-indexer --shell="/bin/bash" --command="${SECURITY_PATH}/tools/securityadmin.sh -cd ${SECURITY_PATH}/securityconfig -cacert ${CONFIG_PATH}/certs/root-ca.pem -cert ${CONFIG_PATH}/certs/admin.pem -key ${CONFIG_PATH}/certs/admin-key.pem -h ${HOST} -p ${PORT} ${OPTIONS}"
+    getPort
+
+    if [ ! -f "${CONFIG_FILE}" ]; then
+        echo "ERROR: it was not possible to find ${CONFIG_FILE}"
+        exit 1
```
Comment (okynos) [inline stack/indexer/indexer-security-init.sh:64]: Move to the Line 10

**Review by okynos — CHANGES_REQUESTED** [review]
Minor changes

File: stack/indexer/indexer-security-init.sh (hunk 5 above)
Comment (miguelfdez99) [reply stack/indexer/indexer-security-init.sh]: Done https://github.com/wazuh/wazuh-packages/pull/1441/commits/c88e0779f55f2646faab03b81d4c9a9e5a7fa551

**Review by alberpilot — APPROVED** [review]

File: stack/indexer/indexer-security-init.sh [hunk 6]
```diff
@@ -8,44 +8,88 @@
...
-
-    # Allow to find ip with an interface
-    if [[ "${HOST}" =~ _.*_ ]]; then
-        interface="${HOST//_}"
-        HOST=$(ip -o -4 addr list ${interface} | awk '{print $4}' | cut -d/ -f1)
-    fi
 
-    if [ "${HOST}" = "0.0.0.0" ]; then
-        HOST="127.0.0.1"
+    if [ -f "${CONFIG_FILE}" ]; then
```
Comment (okynos) [inline stack/indexer/indexer-security-init.sh:43]: Not needed

File: stack/indexer/indexer-security-init.sh [hunk 7]
```diff
@@ -8,44 +8,88 @@
...
+            HOST="127.0.0.1"
+        fi
     fi
+
 }
 
 # -----------------------------------------------------------------------------
+getPort() {
+
+    if [ -f "${CONFIG_FILE}" ]; then
```
Comment (okynos) [inline stack/indexer/indexer-security-init.sh:65]: Not needed

**Review by okynos — CHANGES_REQUESTED** [review]
Minor change, GJ

**Review by okynos — APPROVED** [review]
LGTM

**Pull request merged**